csv/src/fs2/data/csv/CsvRow.scala

```diff
@@ -23,13 +23,115 @@ class CsvRow[Header](val values: NonEmptyList[String], val headers: Option[NonEm
   private lazy val byHeader: Option[Map[Header, String]] =
     headers.map(_.toList.zip(values.toList).toMap)
 
+  /** Number of cells in the row. */
+  def size: Int = values.size
+
+  /** Returns the content of the cell at `idx` if it exists.
+    * Returns `None` if `idx` is out of row bounds.
+    * An empty cell value results in `Some("")`.
+    */
   def apply(idx: Int): Option[String] =
     values.get(idx)
 
+  /** Modifies the cell content at the given `idx` using the function `f`.
+    */
+  def modify(idx: Int)(f: String => String): CsvRow[Header] =
+    if (idx < 0 || idx >= values.size)
+      this
+    else
+      new CsvRow(values.zipWithIndex.map {
+        case (cell, i) =>
+          if (i === idx)
+            f(cell)
+          else
+            cell
+      }, headers)
+
+  /** Modifies the cell content at the given `header` using the function `f`.
+    */
+  def modify(header: Header)(f: String => String): CsvRow[Header] =
```

Comment:
The semantics of these methods depend on all `header` values being unique (as it will only modify the first occurrence). Not sure there is actually a convincing use case for having the same value for several columns which would justify changing the current semantics. But at least I suggest documenting it clearly.

Reply:
This is true. I'll comment it.